### PR TITLE
Feature/idcom 2186  cli implementation

### PR DESCRIPTION
**Pull request by DiosAkhilleus**

### CODE REVIEW

File: solana/program_v2/programs/gateway_v2/src/instructions/admin/close_network.rs [hunk 1]
```diff
@@ -17,7 +18,10 @@ pub struct CloseNetworkAccount<'info> {
     #[account(
         mut,
         close = destination,
-        constraint = network.can_access(&authority, NetworkKeyFlags::AUTH),
+        seeds = [NETWORK_SEED, network.authority.key().as_ref(), &network.network_index.to_le_bytes()],
+        bump,
+        // TODO: @william commented constraint for now
```
Comment (william-brooks) [inline solana/program_v2/programs/gateway_v2/src/instructions/admin/close_network.rs:23]: these constraint issues have been fixed (check below as there are a bunch of these)

File: solana/program_v2/packages/client/cli/src/commands/network/create.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      char: 'k',
+      description: 'Path to a Solana keypair',
+      required: true,
+    }),
+    cluster: Flags.string({
+      char: 'c',
+      description: 'The cluster you wish to use',
+      required: true,
+    }),
+    index: Flags.integer({
```
Comment (william-brooks) [inline solana/program_v2/packages/client/cli/src/commands/network/create.ts:30]: This is no longer needed

File: solana/program_v2/packages/client/core/package.json [hunk 3]
```diff
@@ -25,9 +25,9 @@
         "build": "tsc -p tsconfig.json"
     },
     "dependencies": {
+        "@identity.com/gateway-solana-idl": "^2.0.0-beta.1",
```
Comment (william-brooks) [inline solana/program_v2/packages/client/core/package.json:28]: please update to beta-6

File: solana/program_v2/packages/tests/src/gatekeeper-suite/set-gatekeeper-state.test.ts [hunk 4]
```diff
@@ -91,7 +91,7 @@ describe('Gateway v2 Client', () => {
   });
 
   describe('Set Gatekeeper State', () => {
-    it("Should set a gatekeeper's state", async function () {
+    it.skip("Should set a gatekeeper's state", async function () {
```
Comment (william-brooks) [inline solana/program_v2/packages/tests/src/gatekeeper-suite/set-gatekeeper-state.test.ts:94]: why is this skipped ?

Comment (DiosAkhilleus) [reply solana/program_v2/packages/tests/src/gatekeeper-suite/set-gatekeeper-state.test.ts:94]: @JulianSpring skipped some of these during his testing... Unsure if he meant to leave them skipped?

File: .github/workflows/solana-v2-rust.yml [hunk 5]
```diff
@@ -144,12 +144,12 @@ jobs:
       - name: Install
         run: yarn install --ignore-scripts --frozen-lockfile
 
-      - name: Lint
-        run: yarn lint
-
       - name: Build program
         run: yarn build
 
+      - name: Lint
```
Comment (william-brooks) [inline .github/workflows/solana-v2-rust.yml:150]: I think this was changed for a reason? Or was it breaking?

File: solana/program_v2/programs/gateway_v2/src/instructions/gatekeeper/change_pass_gatekeeper.rs [hunk 6]
```diff
@@ -28,6 +28,7 @@ pub struct PassChangeGatekeeper<'info> {
     #[account(
     seeds = [PASS_SEED, pass.subject.as_ref(), pass.network.key().as_ref(), &pass.pass_number.to_le_bytes() ],
     bump,
+    // TODO: @william commented constraint for now
```
Comment (william-brooks) [inline solana/program_v2/programs/gateway_v2/src/instructions/gatekeeper/change_pass_gatekeeper.rs:31]: could be removed?

File: solana/program_v2/programs/gateway_v2/src/instructions/gatekeeper/pass_set_state.rs [hunk 7]
```diff
@@ -20,6 +20,7 @@ pub struct PassSetState<'info> {
     #[account(
         seeds = [PASS_SEED, pass.subject.as_ref(), pass.network.key().as_ref(), &pass.pass_number.to_le_bytes() ],
         bump,
+        // TODO: @william commented constraint for now
```
Comment (william-brooks) [inline solana/program_v2/programs/gateway_v2/src/instructions/gatekeeper/pass_set_state.rs:23]: could be removed?

File: solana/program_v2/programs/gateway_v2/src/instructions/gatekeeper/verify_pass.rs [hunk 8]
```diff
@@ -58,7 +58,9 @@ pub struct PassVerify<'info> {
     #[account(
     seeds = [PASS_SEED, pass.subject.as_ref(), network.key().as_ref(), & pass.pass_number.to_le_bytes()],
     bump,
-    mut,
+    // TODO: @william commented constraint for now
```
Comment (william-brooks) [inline solana/program_v2/programs/gateway_v2/src/instructions/gatekeeper/verify_pass.rs:61]: This should be working now?

File: solana/program_v2/programs/gateway_v2/src/instructions/gatekeeper/set_pass_data.rs [hunk 9]
```diff
@@ -25,6 +25,7 @@ pub struct PassSetData<'info> {
     #[account(
         seeds = [PASS_SEED, pass.subject.as_ref(), pass.network.key().as_ref(), &pass.pass_number.to_le_bytes() ],
         bump = pass.signer_bump,
+        // TODO: @william commented out constraint for now
```
Comment (william-brooks) [inline solana/program_v2/programs/gateway_v2/src/instructions/gatekeeper/set_pass_data.rs:28]: could be removed?

File: solana/program_v2/programs/gateway_v2/src/instructions/network/close_gatekeeper.rs [hunk 10]
```diff
@@ -29,7 +29,8 @@ pub struct CloseGatekeeperAccount<'info> {
     close = destination,
     seeds = [GATEKEEPER_SEED, authority.key().as_ref(), gatekeeper.gatekeeper_network.key().as_ref()],
     bump = gatekeeper.gatekeeper_bump,
-    constraint = gatekeeper.can_access(& authority, GatekeeperKeyFlags::AUTH),
+    // TODO: @william commented constraint for now
```
Comment (william-brooks) [inline solana/program_v2/programs/gateway_v2/src/instructions/network/close_gatekeeper.rs:32]: could be removed?

File: .github/workflows/solana-v2-rust.yml (hunk 5 above)
Comment (DiosAkhilleus) [reply .github/workflows/solana-v2-rust.yml:150]: Hmm I don't remember changing that so maybe it switched during a backmerge. Will look

File: solana/program_v2/programs/gateway_v2/src/instructions/gatekeeper/verify_pass.rs (hunk 8 above)
Comment (DiosAkhilleus) [reply solana/program_v2/programs/gateway_v2/src/instructions/gatekeeper/verify_pass.rs:61]: added back

File: solana/program_v2/packages/tests/src/gatekeeper-suite/set-gatekeeper-state.test.ts (hunk 4 above)
Comment (JulianSpring) [reply solana/program_v2/packages/tests/src/gatekeeper-suite/set-gatekeeper-state.test.ts:94]: Oh sorry! I will fix this in my current task

Comment (JulianSpring) [reply solana/program_v2/packages/tests/src/gatekeeper-suite/set-gatekeeper-state.test.ts:94]: Not really sure why it is showing as a change though. It is part of feature/gateway_v2

File: solana/program_v2/packages/tests/src/network-suite/create-network.test.ts [hunk 11]
```diff
@@ -37,6 +37,13 @@ describe('Gateway v2 Client', () => {
       },
       programProvider
     );
+    //   export declare type GatewayServiceOptions = {
```
Comment (JulianSpring) [inline solana/program_v2/packages/tests/src/network-suite/create-network.test.ts:40]: Think we can remove these commented out lines?

Comment (DiosAkhilleus) [reply solana/program_v2/packages/tests/src/network-suite/create-network.test.ts:40]: I'd think so!

File: solana/program_v2/packages/tests/src/gatekeeper-suite/set-gatekeeper-state.test.ts (hunk 4 above)
Comment (william-brooks) [reply solana/program_v2/packages/tests/src/gatekeeper-suite/set-gatekeeper-state.test.ts:94]: @JulianSpring @DiosAkhilleus it's fine, leave it then... we can fix it later

**Review by william-brooks — APPROVED** [review]

**Pull request merged**